### ISSUE TITLE
setup.exe

### ISSUE THREAD

**Issue by Audittman** [nanowebcoder/NanoVeraHuesBridge#2]
Hello
When I click on "setup.exe"
![image](https://cloud.githubusercontent.com/assets/18665978/14795102/3b3caa5a-0b27-11e6-9dd1-aebaf160568e.png)

, at the beginning of the configuration, I can not access the configuration window as you mention ... but i have this :
![image](https://cloud.githubusercontent.com/assets/18665978/14795082/2172ec1a-0b27-11e6-8e81-d631aaf3b744.png)

I work with VMWareFusion (windows 7) for MAC.

**Comment by nanowebcoder** [comment]
Instructions could be clearer (and github could be easier to use).  Click the "View Raw," then choose to save the file.  You'll have to do this for the 2 files in the Setup directory (setup.exe and the NanoVeraHuesBridge.msi file).  Then, from your computer, where you downloaded the setup files, run the setup.exe.

Github doesn't make it easy to download a file or two.

**Comment by Audittman** [comment]
Hello

thanks a lot for your help. I could move forward.

**Comment by Audittman** [comment]
Hello
I meet the other problem now : I said that I am in France and I bought the ECHO in the US. and that may be the problem is that SSDP may not be the same in France ... I don't really know ...
Here is the error that occurs when right click on the service and choosing " start" . And the "LOG" file.

[log.txt](https://github.com/nanowebcoder/NanoVeraHuesBridge/files/242112/log.txt)

![message erreur](https://cloud.githubusercontent.com/assets/18665978/14913532/fc0aad52-0e03-11e6-8a66-129c7a00e6c2.png)

![local ipport](https://cloud.githubusercontent.com/assets/18665978/14913535/010f9f88-0e04-11e6-96e0-3c5d19e1ec71.png)

**Comment by nanowebcoder** [comment]
My French is not nearly as good as your English, but it looks like the log file doesn’t like the IP address or the port

2016-04-29 11:59:24.4820 | VeraHuesBridge.SSDPService | WARN | Error occured starting SSDP service. | L’adresse demandée n’est pas valide dans son contexte

The things I would check are:

a)      Is 192.168.1.100 the correct IP address (should be the IP of the computer running the bridge, not the echo)

b)      Anything else using port 1900?  You might need to stop the SSDP Discovery service if it is running.  I have heard people say it can interfere with other software on the machine.  Or try a different port (1901 for instance)

**Comment by Audittman** [comment]
Hello
Thank you very much for using valuable assistance . it helped me to understand a little better how this utility works ...
Remains to configure my devices.

**Comment by Audittman** [comment]
Hello.
I get to the part where I ask Alexa to find the devices.
Here are my DeviceConfig & Log files.

[DeviceConfig.txt](https://github.com/nanowebcoder/NanoVeraHuesBridge/files/243825/DeviceConfig.txt)

[log.txt](https://github.com/nanowebcoder/NanoVeraHuesBridge/files/243826/log.txt)

![image](https://cloud.githubusercontent.com/assets/18665978/14938530/96cddcc4-0f26-11e6-81e9-562f10ecc052.png)

192.168.1.51 is my IP address of my PC. 

Hue Bridge has been started locally : 
![image](https://cloud.githubusercontent.com/assets/18665978/14938541/0a29b922-0f27-11e6-9a62-2defebaac878.png)

Alexa said she does not find any device when I run the search: I ask myself the question: is it because I 'm emulation with VMWare Fusion and my Alexa is perhaps not recognized on my wifi windows 7 emulated ?
![image](https://cloud.githubusercontent.com/assets/18665978/14938560/a8474bf6-0f27-11e6-8a0b-db846ad7f7cf.png)

**Comment by nanowebcoder** [comment]
The log says its trying to listen on 192.168.1.6 but you say your PC is 192.168.1.51.  Can you list out the ip addresses of your PC, your echo and your Vera?  The. We need to check the application configuration

On Apr 30, 2016, at 5:03 PM, Audittman <notifications@github.com<mailto:notifications@github.com>> wrote:

Hello.
I get to the part where I ask Alexa to find the devices.
Here are my DeviceConfig & Log files.

DeviceConfig.txthttps://github.com/nanowebcoder/NanoVeraHuesBridge/files/243825/DeviceConfig.txt

log.txthttps://github.com/nanowebcoder/NanoVeraHuesBridge/files/243826/log.txt

[image]https://cloud.githubusercontent.com/assets/18665978/14938530/96cddcc4-0f26-11e6-81e9-562f10ecc052.png

192.168.1.51 is my IP address of my PC.

Hue Bridge has been started locally :
[image]https://cloud.githubusercontent.com/assets/18665978/14938541/0a29b922-0f27-11e6-9a62-2defebaac878.png

Alexa said she does not find any device when I run the search: I ask myself the question: is it because I 'm emulation with VMWare Fusion and my Alexa is perhaps not recognized on my wifi windows 7 emulated ?
[image]https://cloud.githubusercontent.com/assets/18665978/14938560/a8474bf6-0f27-11e6-8a0b-db846ad7f7cf.png

##

**Comment by Audittman** [comment]
Hello.

Here is :
- IP of my computer (I said that the IPControl is doing on my windows 7 on my Windows virtual (VMWare ) .
  In fact I do everything on that because I work otherwise on Mac.).
  ![image](https://cloud.githubusercontent.com/assets/18665978/14940929/8c13647a-0f8a-11e6-89d0-bff11b14a8f4.png)

But my Imac IP is : 
![image](https://cloud.githubusercontent.com/assets/18665978/14940997/39c97c06-0f8d-11e6-9812-fab08bde9e9d.png)
- VERA IP : 192.168.1.16
  ![image](https://cloud.githubusercontent.com/assets/18665978/14940944/0e3b779e-0f8b-11e6-9311-406e93c966a6.png)
- ECHO IP : 
  ![image](https://cloud.githubusercontent.com/assets/18665978/14940957/9c823e34-0f8b-11e6-9d4b-6fd45354f835.png)

![image](https://cloud.githubusercontent.com/assets/18665978/14940963/d9c2a4a0-0f8b-11e6-86a6-7ca5be8c4731.png)

**Comment by nanowebcoder** [comment]
In the application configuration, you want to use 192.168.1.51

On May 1, 2016, at 5:00 AM, Audittman <notifications@github.com<mailto:notifications@github.com>> wrote:

Hello.

Here is :
-   IP of my computer (I said that the IPControl is doing on my windows 7 on my Windows virtual (VMWare ) .
  In fact I do everything on that because I work otherwise on Mac.).
  [image]https://cloud.githubusercontent.com/assets/18665978/14940929/8c13647a-0f8a-11e6-89d0-bff11b14a8f4.png
-   VERA IP : 192.168.1.16
  [image]https://cloud.githubusercontent.com/assets/18665978/14940944/0e3b779e-0f8b-11e6-9311-406e93c966a6.png
-   ECHO IP :
  [image]https://cloud.githubusercontent.com/assets/18665978/14940957/9c823e34-0f8b-11e6-9d4b-6fd45354f835.png

[image]https://cloud.githubusercontent.com/assets/18665978/14940963/d9c2a4a0-0f8b-11e6-86a6-7ca5be8c4731.png

##

**Comment by Audittman** [comment]
Sorry, I don't understand your answer ..

**Comment by Audittman** [comment]
When you say in the "Getting.Started.with.Nano.Vera.Hues.Bridge.pdf":

"However the Amazon Echo will not know about new devices until you tell Vera to Discover Devices."
what do you mean ?
I have already devices in my VERA. I have to make a manipulation in my VERA ? I do not see how I can detect the ECHO BRIDGE HUE . Because it is on the same network and the service Nano Vera/Hues Bridge is started ? Where is that ECHO will search for devices and created with DeviceEditor.exe ?

Thank you for the time spent trying to solve my problems : I really want this to work at home ...

**Comment by nanowebcoder** [comment]
I have a typo it appears:

"However the Amazon Echo will not know about new devices until you tell Vera to Discover Devices."

Should read:

"However the Amazon Echo will not know about new devices until you tell Echo to Discover Devices."

I am saying you need to literally tell Echo to “Discover New Devices”

Echo will then perform a uPNP discovery of your network.  If the NanoVeraHuesBridge is running and setup correctly, Echo will discover it.   This means Echo will then ask the bridge to tell it what devices it has configured.  Echo will then say “I have discovered X number of new devices).  If it were a conversation it would look like this:

You: Echo, discover new devices.
Echo: Ok….
Echo (to your network): Hey anybody out there?
Bridge:  Yes I am here.  Hello.
Echo (to the bridge): What devices do you control?
Bridge: I control X, Y and Z devices.
Echo (back to you): I found 3 new devices.

**Comment by Audittman** [comment]
Yes, i'd  literally tell Echo to “Discover New Devices”...
However ECHO replies that he can not find anything ... That's my problem now.

**Comment by nanowebcoder** [comment]
Need to make sure you have the right IP and ports assigned.  Let’s get a fresh log:

1)      Stop the service

2)      Delete the log file

3)      Start the service

4)      Tell Echo to discover new devices.

5)      When done, stop the service

6)      Send the log file

Also send a copy of the application configuration (NanoVeraHuesBridgeService.exe.config)

**Comment by Audittman** [comment]
OK thanks. I'll do it tonight when I get home from work ...
Thanks again

**Comment by Audittman** [comment]
Send the log file:
[log.txt](https://github.com/nanowebcoder/NanoVeraHuesBridge/files/247825/log.txt)

Also send a copy of the application configuration (NanoVeraHuesBridgeService.exe.config):
![image](https://cloud.githubusercontent.com/assets/18665978/14999401/54605410-1189-11e6-910e-2b638add96f3.png)

**Comment by nanowebcoder** [comment]
Looks like it was working fine, however it appears the system was stopping just as it received the uPNP from the echo.  The last line was “Received a UDP multicast…”, but it came in after the “Service is down”

Hate to make you do it again, but we might need to wait a little while (don’t stop the service until Echo says she has completed device discovery).

2016-05-03 23:42:10.5455 | VeraHuesBridge.SSDPService | INFO | Stopping SSDP Service... |
2016-05-03 23:42:10.5455 | VeraHuesBridge.SSDPService | INFO | SSDP Service stopped. |
2016-05-03 23:42:10.5455 | NanoVeraHuesBridgeService.NanoVeraHuesBridgeService | INFO | OnStop: SSDP listener is down. |
2016-05-03 23:42:10.5455 | VeraHuesBridge.WebServer | INFO | Webserver stopping... |
2016-05-03 23:42:10.5455 | VeraHuesBridge.WebServer | INFO | Webserver stopped. |
2016-05-03 23:42:10.5455 | NanoVeraHuesBridgeService.NanoVeraHuesBridgeService | INFO | OnStop: Webserver is down. |
2016-05-03 23:42:10.5455 | NanoVeraHuesBridgeService.NanoVeraHuesBridgeService | INFO | OnStop: Service is down. |
2016-05-03 23:42:10.5455 | VeraHuesBridge.SSDPService | INFO | Received a UDP multicast from IP [239.255.255.250], on port [1900]. |

**Comment by Audittman** [comment]
hi
I didn't changed this:
![image](https://cloud.githubusercontent.com/assets/18665978/15009442/a7c51b24-11e6-11e6-863e-46f3a7352f64.png)

Is that correct? As with any place you mentioned that we should change ...
Is 3480 the good PORT in DeviceConfig text ?
In DeviceEditor, i only changed the vera adress with mine, ans DeviceNum with the number of my devices .... Correct ?

Here is my LOG file:
[log.txt](https://github.com/nanowebcoder/NanoVeraHuesBridge/files/248454/log.txt)

Thanks.

**Comment by nanowebcoder** [comment]
You typically do not need to change the multicast IP unless your network topology requires it (and there’s no way I can tell you what to change it to—you’d have to know your topology).  The multicast port only needs be changed if something else is using that port.  So those look right.

And yes, in DeviceConfig, you replace the DeviceNums and IP address and Port of your vera to match your Vera.  Port 3480 is typically what Vera runs on, so unless you changed it, it will be that port.  You can test these by simply putting the onUrl (for instance) into your browser.  It should turn on whatever device its configured to control.  (And the offURL should shut it off).  If it doesn’t work, you need to check your Vera to figure out that information.

But our problem isn’t the DeviceConfig.  The problem is that Echo hasn’t discovered the devices yet (so we aren’t even getting to the device config stuff).

This log shows the same problem as before (namely its getting a UDP multicast AFTER the system shuts down).  Nothing appears to be happening when the system is running.  There are no entries between System up and Service down.  That is where I expect to see a UDP multcast coming from your Vera..

2016-05-04 11:10:13.0734 | NanoVeraHuesBridgeService.NanoVeraHuesBridgeService | INFO | OnStart: WS started. System is up. |

(NOTHING HERE?)

2016-05-04 11:11:56.5256 | NanoVeraHuesBridgeService.NanoVeraHuesBridgeService | INFO | OnStop: Service is shutting down... |

Regrettably, we’re going to have turn up logging to its highest level and repeat the procedure.  Here’s how to turn up the logging level:

1)      Stop the service

2)      Edit the NLog.config file with a text editor like Notepad

a.       Under the <Rules> section, change this line:

<logger name="*" minlevel="Info" writeTo="logfile" />
                                To
<logger name="*" minlevel="Trace" writeTo="logfile" />

3)      Start the service (give it a minute to get ready)

4)      Tell Echo to discover new devices

5)      Wait a good 5 minutes

6)      Stop service

7)      Submit the log.

**Comment by Audittman** [comment]
What is OK for me is the format, in my browser:
"onUrl" : "http://192.168.1.16:3480/data_request?id=action&output_format=json&serviceId=urn:upnp-org:serviceId:SwitchPower1&action=SetTarget&newTargetValue=1&DeviceNum=5",
  "offUrl" : "http://192.168.1.16:3480/data_request?id=action&output_format=json&serviceId=urn:upnp-org:serviceId:SwitchPower1&action=SetTarget&newTargetValue=0&DeviceNum=5"

and not your format in Nano Vera/Hues Bridge Device Editor:
onUrl : http://192.168.1.16:3480/data_request?id=action&output_format=json&DeviceNum=5&serviceId=urn:upnp-org:serviceId:Dimming1&action=SetLoadLevelTarget&newLoadlevelTarget=${intensity.percent}

OffUrl : http://192.168.1.16:3480/data_request?id=action&output_format=json&serviceId=urn:upnp-org:serviceId:SwitchPower1&action=SetTarget&newTargetValue=0&DeviceNum=5

In fact, your OnUrl or OffUrl turn off my lights when they are ON. When my lights are OFF, your offUrl or OnUrl have no effect ...

Here my Log file 
[log.txt](https://github.com/nanowebcoder/NanoVeraHuesBridge/files/253622/log.txt)

**Comment by nanowebcoder** [comment]
Your onUrl is a switch command for a light that can only be turned on or off (it’s not dimmable).  The newTargetValue=1 means to turn it on.  newTargetValue=0 means turn it off.

The format shown in the device editor is for a dimmable capable light (hence the call to newLoadlevelTarget=intensity.percent).)

If your switch is not dimmable:
And you use the device editor’s format, when you say to Alexa, “turn bedroom lamp on”, then it won’t do anything.

If your switch is dimmable
Then it still wont do anything unless the “DefaultIntensity” setting in the NanoVeraHuesBridgeService.exe.config file is set high enough to turn the light on.  You could say, instead, “Alexa, set bedroom lamp to 90 percent” and it will turn it on.
